src/neo/IO/Json/JNumber.cs

```diff
@@ -14,6 +13,7 @@ public class JNumber : JObject
 
         public JNumber(double value = 0)
         {
+            if (!double.IsFinite(value)) throw new FormatException();
             this.Value = value;
```

Comment:
We should check 
```
if (value > MAX_SAFE_INTEGER || value < MIN_SAFE_INTEGER)
                throw new FormatException();
```
?

Reply:
I need to take another look at the JSON specification.

Reply:
We don't need to check that. Just convert the number to `double`, and see if it is finite.

Reply:
We should remove this `const`?

Reply:
It was used in

Reply:
Because integers outside this range cannot be deserialized without loss of precision after they are serialized.